[docs]: add documentation for WebMCP (#2262)
browserbase/stagehand: packages/docs/v3/basics/webmcp.mdx

console.log(tools);

await stagehand.close();
```

Each tool includes its `name`, optional `description`, optional `inputSchema`, optional `annotations`, and the `frameId` where it was registered.

<Note>
`listWebMCPTools()` returns a fresh snapshot from the browser on each call. It does not reuse tools from previous pages or previous calls.
</Note>

## Invoking Tools

Use `page.invokeWebMCPTool()` with the tool name and a JSON-serializable input object. The method returns an invocation immediately; await `invocation.result` for the tool's final response.

```typescript
const tools = await page.listWebMCPTools();
const flightSearch = tools.find((tool) => tool.name === "searchFlights");

if (!flightSearch) {
  throw new Error("searchFlights was not registered on this page");
}

const invocation = await page.invokeWebMCPTool(
  flightSearch.name,
  {
    origin: "SFO",
    destination: "JFK",
    tripType: "round-trip",
    outboundDate: "2026-06-10",
    inboundDate: "2026-06-17",
    passengers: 1,
  },
  { frameId: flightSearch.frameId },
);

const result = await invocation.result;
console.log(result.status);
console.log(result.output);
```

<Tip>

> Pass `frameId` from `listWebMCPTools()` when invoking a tool. If you omit it, Stagehand resolves the frame automatically only when exactly one registered tool has that name.

Reviewer: just a nit:

```suggestion
Pass `frameId` from `listWebMCPTools()` when invoking a tool. If omitted, Stagehand resolves the frame automatically, but only if exactly one registered tool has that name.
```
